scvi/data/_built_in_data/_cite_seq.py

```diff
@@ -143,3 +145,80 @@ def _load_spleen_lymph_cite_seq(
         )
 
     return dataset
+
+
+def _load_pbmc_seurat_v4_cite_seq(
+    save_path: str = "data/",
+    apply_filters: bool = True,
+    aggregate_proteins: bool = True,
+    mask_protein_batches: int = 0,
+    run_setup_anndata: bool = True,
+):
+    url = "https://ndownloader.figshare.com/files/27458840"
+    save_fn = "pbmc_seurat_v4.h5ad"
+    _download(url, save_path, save_fn)
+    adata = anndata.read_h5ad(os.path.join(save_path, save_fn))
+
+    if aggregate_proteins is True:
+        protein_df = pd.DataFrame(index=adata.obsm["protein_counts"].index)
+        ref_proteins = adata.obsm["protein_counts"].columns
+        for p in ref_proteins:
+            if p.split("-")[-1] == "1" or p.split("-")[-1] == "2":
+                root = p.split("-")[0]
+                if root not in ["Notch", "TCR"]:
+                    try:
+                        protein_df[root] = (
+                            adata.obsm["protein_counts"][root + "-1"]
+                            + adata.obsm["protein_counts"][root + "-2"]
+                        ).values
+                    except KeyError:
+                        protein_df[p] = adata.obsm["protein_counts"][p]
+                else:
+                    protein_df[p] = adata.obsm["protein_counts"][p]
+            else:
+                protein_df[p] = adata.obsm["protein_counts"][p]
+        adata.obsm["protein_counts"] = protein_df
+
+    if apply_filters:
+        adata.obs["total_counts"] = np.ravel(adata.X.sum(axis=1).A)
+        adata.var["mt"] = adata.var_names.str.startswith("MT-")
+        adata.obs["total_counts_mt"] = np.ravel(
+            adata.X[:, adata.var["mt"].values].sum(axis=1).A
+        )
+        adata.obs["pct_counts_mt"] = (
+            adata.obs["total_counts_mt"] / adata.obs["total_counts"] * 100
+        )
+
+        adata.obs["Protein log library size"] = np.log(
+            adata.obsm["protein_counts"].sum(1)
+        )
+        adata.obs["Number proteins detected"] = (adata.obsm["protein_counts"] > 0).sum(
+            1
+        )
+        adata.obs["RNA log library size"] = np.log(adata.X.sum(1).A)
+
+        # actually filter
+        adata = adata[adata.obs["Protein log library size"] > 7.6]
+        adata = adata[adata.obs["Protein log library size"] < 10.3]
+        adata = adata[adata.obs["Number proteins detected"] > 150]
+        # filter doublet
+        adata = adata[adata.obs["celltype.l2"] != "Doublet"]
+        # MT
+        adata = adata[adata.obs["pct_counts_mt"] < 12].copy()
```

Comment:
Why do we make a copy here?

Reply:
with anndata, without the copy that will be a view of the object in memory, which is not compatible with setup anndata

Reply:
ahhh.
Curiosity question: is this copy an expensive copy the same way as any `adata2 = adata.copy()` is (where adata is _not_ a view)? Or is it a call that doesn't involve any actual byte transfer, but instead just marks the view as a full adata?

Reply:
this is a good question.... and I don't have the exact answer